go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -53,27 +54,65 @@ func podToCookie(pod *kapi.Pod) string {
 	return nameToCookie(pod.Namespace + "_" + pod.Name)
 }
 
-func (n *NodeController) addOrUpdatePod(pod *kapi.Pod) error {
+func (n *NodeController) waitForNamespace(name string, wf *factory.WatchFactory) (*kapi.Namespace, error) {
+	var namespaceBackoff = wait.Backoff{Duration: 1 * time.Second, Steps: 7, Factor: 1.5, Jitter: 0.1}
+	var namespace *kapi.Namespace
+	if err := wait.ExponentialBackoff(namespaceBackoff, func() (bool, error) {
+		var err error
+		namespace, err = wf.GetNamespace(name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Namespace not found; retry
+				return false, nil
+			}
+			klog.Warningf("error getting namespace: %v", err)
+			return false, err
+		}
+		return true, nil
+	}); err != nil {
+		return nil, fmt.Errorf("failed to get namespace object: %v", err)
+	}
+	return namespace, nil
+}
+
+func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, wf *factory.WatchFactory) error {
 	podIPs, podMAC, err := getPodDetails(pod, n.nodeName)
 	if err != nil {
 		klog.V(5).Infof("cleaning up hybrid overlay pod %s/%s because %v", pod.Namespace, pod.Name, err)
 		return n.deletePod(pod)
 	}
+	portMAC, portIP, _ := util.GetPortAddresses(houtil.GetHybridOverlayPortName(n.nodeName))
```

Comment:
@rcarrillocruz you already have portMAC: `n.drMAC`. And you know the IP with `GetNodeHybridOverlayIfAddr()`. We don't need to read the OVS port for it again.

Reply:
Also, is portIP even used in this function?

Reply:
sweet, thx